include/turtle/detail/move_helper.hpp

```diff
@@ -0,0 +1,79 @@
+// http://turtle.sourceforge.net
+//
+// Copyright Mathieu Champlon 2018
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef MOCK_MOVE_HELPER_HPP_INCLUDED
+#define MOCK_MOVE_HELPER_HPP_INCLUDED
+
+#include "../config.hpp"
+#include <boost/type_traits/conditional.hpp>
+#include <boost/type_traits/is_reference.hpp>
+#include <boost/type_traits/remove_reference.hpp>
+#include <boost/type_traits/add_reference.hpp>
+#include <boost/type_traits/add_rvalue_reference.hpp>
+#include <boost/type_traits/decay.hpp>
+
+namespace mock
+{
+namespace detail
+{
+#ifdef MOCK_RVALUE_REFERENCES
+    template< typename T >
+    struct forward_type
+    {
+        typedef T type;
+    };
+    template< typename T >
+    struct ref_arg
+    {
+        typedef typename boost::conditional<
+          boost::is_reference<T>::value,
+          T,
+          typename boost::add_rvalue_reference<T>::type >::type type;
+    };
+
+    template< typename T >
+    inline T&& move_if_not_lvalue_reference(typename boost::remove_reference<T>::type& t)
```

Comment:
I don't think `inline` is needed or even brings any value for template functions.

Reply:
Aha I actually had read that SO question and I even upvoted the comment that says `The explicit specialization is not a template.` :)
I apparently believed the accepted answer was wrong also, as I had upvoted the other one as well… 🤔

Reply:
A hint to what? That we reeeeally would like you to optimize the function call away please Mr compiler ? :)
Anyway it's no big deal and won't break user code, so we can leave it or remove it, whichever…